### PR TITLE
Change 403 error message when enforcer is unavailable

### LINKED ISSUE #1831: If enforcer is down router returns 403 Forbidden
### Description:
If router can not connect with enforcer for the external authz gRPC call, then router will send 403 Forbidden http status code to the client. Ideally we should sent the http status code 500, and the error code 900900 for uncalssified authentication failure

### Environment details (with versions):
- OS:
- Client:
- Env (Docker/K8s):

---

### PR DESCRIPTION
**Pull request by praminda**
### Purpose

When enforcer was down while router is serving requests, router was sending a `403` response to the client without a body. In reality this scenario is not a `403` but a `500` where choreo connect component communication failed. Therefore added a `LocalReply` config in envoy configuration to customize the response when enforcer connection is broken. This error is detected by `UAEX` response flag as documented in [response-flags](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage#config-access-log-format-response-flags)
### Issues

Fixes #1831

### Automation tests
 - Unit tests added: No
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

**Review by VirajSalaka — APPROVED** [review]

**Comment by Rajith90** [comment]
Does this gives the error message when there was an exception thrown in the enforcer side while validating the request. Because on those scenarios also router sends 403

**Comment by praminda** [comment]
@Rajith90 Yes. Just checked by intentionally throwing a NPE from enforcer. This is returned to the client.
```
< HTTP/1.1 500 Internal Server Error
< content-length: 118
< content-type: application/json
< server: envoy
<
{"code":"900900","description":"Error during validating the request","message":"Unclassified Authentication Failure"}
```

File: adapter/internal/oasparser/envoyconf/response_mapper.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	access_logv3 "github.com/envoyproxy/go-control-plane/envoy/config/accesslog/v3"
+	corev3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	hcmv3 "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	"google.golang.org/protobuf/types/known/structpb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+)
+
+func getErrorResponseMappers() []*hcmv3.ResponseMapper {
+	return []*hcmv3.ResponseMapper{
+		genErrorResponseMapper(900900, "Unclassified Authentication Failure",
```
Comment (Rajith90) [inline adapter/internal/oasparser/envoyconf/response_mapper.go:32]: Shall we make this `Unclassified Validation Failure` and also shall we use a different error code as well

**Comment by praminda** [comment]
Sample response of %LOCAL_REPLY_BODY%

```
< HTTP/1.1 500 Internal Server Error
< content-length: 165
< content-type: application/json
< date: <...>
< server: envoy
<
{"message":"Upstream connection failed","code":"102503","description":"upstream connect error or disconnect/reset before headers. reset reason: connection failure"}
```

**Pull request merged**